hw/ip/csrng/lint/csrng.vlt

```diff
@@ -4,3 +4,8 @@
 //
 // waiver file for csrng
 
+`verilator_config
+
+// The tool erroneously thinks there would be circular logic through the aes_mux_sel_buf_chk units inside aes_cipher_control.sv.
+// If these units are not generated inside a for loop, everything is fine.
+lint_off -rule UNOPTFLAT -file "*/rtl/aes_cipher_control.sv" -match "Signal unoptimizable: Feedback to clock or circular logic: '*sp2v_sig'"
```

Comment:
Verilator is getting confused about `sp2v_sig` because it bundles up several signals which depend on one another. If we waive the warning, it will work, but Verilator will have to iterate at simulation time to get to a fixpoint. It's maybe better to add a `/* verilator split_var */` comment to the declaration of the variable, which tells Verilator to schedule the bits separately. (I just checked, and this works)

Reply:
Thanks @rswarbrick, this sounds like a better approach. It probably also faster in simulation. However, I think it would be better to not have tool-specific code/pragmas in the RTL. (Part of this PR also removes tool-specific inline lint waivers.) So far we have such Verilator comments only inside:
- hw/ip/prim_generic/rtl/prim_generic_clock_gating.sv: sounds okay to me as it's just a clock gate used in simulation only
- hw/ip/prim/rtl/prim_subst_perm.sv
- hw/ip/prim/rtl/prim_prince.sv

Do we maybe want to formulate some guidance around this? I couldn't find anything in the style guide.

Reply:
The general guidance is to avoid tool-specific "magic comments" where possible in source files, and use configuration files instead. Focus is on "where possible". For split_var, it is possible in a verilator config file, which is effectively our lint config file:

``` 
split_var [-module "<modulename>"] [-task "<taskname>"] -var "<varname>"
split_var [-module "<modulename>"] [-function "<funcname>"] -var "<varname>"
```

https://www.veripool.org/wiki/verilator/Manual-verilator#CONFIGURATION-FILES

Reply:
Neat! Sounds like a good approach to me.

Reply:
Alright, that's done. Simulation speed increased by roughly 20% in my AES-only Verilator simulation.

Reply:
Wow, that's impressive and something to communicate more widely. Would one of you (Pirmin or Rupert) mind having a single slide with an example case for the next silicon meeting, or alternatively send around an email?

Reply:
Meh, I think we should just fix all these warnings and then disallow introducing new ones (I'm working my way slowly down the list!). The performance implications are written down pretty explicitly in the Verilator manual. My impression is that anyone who cares will already know and anyone who doesn't won't want to hear it.